scimath/units/unit_array.py

```diff
@@ -471,6 +477,22 @@ def as_units(self, new_units):
 
         return result
 
+    ### Unit Conversion ########################################################
+
+    def _get_values_base_str(self):
+        """ Build a string representation of the array values.
+        """
+        # this is a little more complicated than it should be in order
+        # to be more resilient to changes to numpy ndarray API
+        base_str = numpy.ndarray.__repr__(self.view(numpy.ndarray))
+        start = base_str.find('(')
+        end = base_str.rfind(')')
+
+        if start > -1 and end > -1:
+            base_str = base_str[start+1:end]
+
+        return base_str
+
```

Comment:
nit: I think a regex match makes the intent clearer.. For e.g.
```
VALUE_PATTERN = re.compile('.*(?P<value>\[.*\]).*')
...
base_str ="{!r}".format(self.view(numpy.ndarray))
matches = VALUE_PATTERN.match(base_str)
if matches:
    base_str = base_str['value']
return base_str
```

Reply:
I also suggest the regexp solution, it's a bit more robust and clear.

Reply:
> it's a bit more robust and clear

I'd have to disagree with this: it's clear that the non-regexp solution is finding the first opening parenthesis and the last closing parenthesis - i.e., that it's finding a pair of matching parentheses. With the regex solution (presuming that @senganal meant something like `re.compile(r'.*(?P<value>\(.*\)).*')`), it's not clear at a glance that we're finding a matching pair of parentheses at all, and if for whatever reason the representation of the array contains inner parentheses, we'll end up with the wrong thing.

```
>>> import numpy as np
>>> dt = np.dtype([('x', float), ('y', float)])
>>> x = np.zeros(2, dtype=dt)
>>> repr(x)
"array([(0.0, 0.0), (0.0, 0.0)], \n      dtype=[('x', '<f8'), ('y', '<f8')])"
>>> import re
>>> re.compile(r'.*(?P<value>\(.*\)).*').match(repr(x)).group('value')
'(0.0, 0.0)'
```

Reply:
Right. good as is then.